crates/runtime/src/vmcontext.rs

```diff
@@ -594,66 +593,47 @@ mod test_vmcaller_checked_anyfunc {
+                ),
+            )*
+        }
 
-impl VMBuiltinFunctionsArray {
-    pub const fn len() -> usize {
-        BuiltinFunctionIndex::builtin_functions_total_number() as usize
-    }
-
-    pub fn initialized() -> Self {
-        use crate::libcalls::*;
-
-        let mut ptrs = [0; Self::len()];
-
-        ptrs[BuiltinFunctionIndex::memory32_grow().index() as usize] =
-            wasmtime_memory32_grow as usize;
-        ptrs[BuiltinFunctionIndex::table_copy().index() as usize] = wasmtime_table_copy as usize;
-        ptrs[BuiltinFunctionIndex::table_grow_funcref().index() as usize] =
-            wasmtime_table_grow as usize;
-        ptrs[BuiltinFunctionIndex::table_grow_externref().index() as usize] =
-            wasmtime_table_grow as usize;
-        ptrs[BuiltinFunctionIndex::table_init().index() as usize] = wasmtime_table_init as usize;
-        ptrs[BuiltinFunctionIndex::elem_drop().index() as usize] = wasmtime_elem_drop as usize;
-        ptrs[BuiltinFunctionIndex::memory_copy().index() as usize] = wasmtime_memory_copy as usize;
-        ptrs[BuiltinFunctionIndex::memory_fill().index() as usize] = wasmtime_memory_fill as usize;
-        ptrs[BuiltinFunctionIndex::memory_init().index() as usize] = wasmtime_memory_init as usize;
-        ptrs[BuiltinFunctionIndex::data_drop().index() as usize] = wasmtime_data_drop as usize;
-        ptrs[BuiltinFunctionIndex::drop_externref().index() as usize] =
-            wasmtime_drop_externref as usize;
-        ptrs[BuiltinFunctionIndex::activations_table_insert_with_gc().index() as usize] =
-            wasmtime_activations_table_insert_with_gc as usize;
-        ptrs[BuiltinFunctionIndex::externref_global_get().index() as usize] =
-            wasmtime_externref_global_get as usize;
-        ptrs[BuiltinFunctionIndex::externref_global_set().index() as usize] =
-            wasmtime_externref_global_set as usize;
-        ptrs[BuiltinFunctionIndex::table_fill_externref().index() as usize] =
-            wasmtime_table_fill as usize;
-        ptrs[BuiltinFunctionIndex::table_fill_funcref().index() as usize] =
-            wasmtime_table_fill as usize;
-        ptrs[BuiltinFunctionIndex::memory_atomic_notify().index() as usize] =
-            wasmtime_memory_atomic_notify as usize;
-        ptrs[BuiltinFunctionIndex::memory_atomic_wait32().index() as usize] =
-            wasmtime_memory_atomic_wait32 as usize;
-        ptrs[BuiltinFunctionIndex::memory_atomic_wait64().index() as usize] =
-            wasmtime_memory_atomic_wait64 as usize;
-        ptrs[BuiltinFunctionIndex::out_of_gas().index() as usize] = wasmtime_out_of_gas as usize;
-        ptrs[BuiltinFunctionIndex::new_epoch().index() as usize] = wasmtime_new_epoch as usize;
-
-        if cfg!(debug_assertions) {
-            for i in 0..ptrs.len() {
-                debug_assert!(ptrs[i] != 0, "index {} is not initialized", i);
+        impl VMBuiltinFunctionsArray {
+            pub fn new() -> &'static Self {
```

Comment:
Rather than `new()`, can we call this `static_instance()` or something like that? As written it was somewhat concerning to read `VMBuiltinFunctionsArray::new()` in the initialization path -- it looked like an allocation of an owned thing.

Reply:
Aha looks like I can do one better and have `const INIT: VMBuiltinFunctionsArray = ...`!